src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -26,48 +26,74 @@ import {quantizeScale, linearScale} from '../../../utils/scale-utils';
 import {defaultColorRange} from '../../../utils/color-utils';
 import {pointToHexbin} from './hexagon-aggregator';
 
+import SortedBins from '../../../utils/sorted-bins';
+
 const defaultProps = {
   colorDomain: null,
   colorRange: defaultColorRange,
   elevationDomain: null,
   elevationRange: [0, 1000],
   elevationScale: 1,
+  lowerPercentile: 0,
+  upperPercentile: 100,
   radius: 1000,
   coverage: 1,
   extruded: false,
   hexagonAggregator: pointToHexbin,
   getPosition: x => x.position,
   fp64: false
-  //
 };
 
 function _needsReProjectPoints(oldProps, props) {
   return oldProps.radius !== props.radius;
 }
 
-function _getCountRange(hexagons) {
-  return [
-    Math.min.apply(null, hexagons.map(bin => bin.points.length)),
-    Math.max.apply(null, hexagons.map(bin => bin.points.length))
-  ];
+function _percentileChanged(oldProps, props) {
+  return oldProps.lowerPercentile !== props.lowerPercentile ||
+    oldProps.upperPercentile !== props.upperPercentile;
 }
 
 export default class HexagonLayer extends Layer {
   constructor(props) {
     if (!props.radius) {
-      log.once(0, 'PointDensityHexagonLayer: radius in meter is needed to aggregate points into ' +
+      log.once(0, 'HexagonLayer: radius in meter is needed to aggregate points into ' +
         'hexagonal bins, Now using 1000 meter as default');
 
       props.radius = defaultProps.radius;
     }
 
+    if (Number.isFinite(props.upperPercentile) &&
```

Comment:
I agree we should have a more formal way of checking props. At the moment, this can't be called by checkRequiredProps. because they are not required props. I only check if those props are valid when user provide them.

Maybe we should create a ticket for better prop checking?